src/Data/Foldable.purs

```diff
@@ -449,10 +447,13 @@ minimum = minimumBy compare
 -- | function. The comparison function should represent a total ordering (see
 -- | the `Ord` type class laws); if it does not, the behaviour is undefined.
 minimumBy :: forall a f. Foldable f => (a -> a -> Ordering) -> f a -> Maybe a
-minimumBy cmp = foldl min' Nothing
+minimumBy cmp = foldlMaybe \x y -> if cmp x y == LT then x else y
+
+foldlMaybe :: forall f a . Foldable f => (a -> a -> a) -> f a -> Maybe a
```

Comment:
I think this should have some documentation? Also, to be more consistent with the rest of the file:

```suggestion
foldlMaybe :: forall f a. Foldable f => (a -> a -> a) -> f a -> Maybe a
```

Reply:
Considering that visible type applications will likely come out sometime in v0.15.x, we should keep the `f` as the first parameter since type applications are order-specific.

Reply:
Yeah, but that should be done in a PR separate from this one.